lib/wickedbase.pm

```diff
@@ -1305,9 +1305,9 @@ sub need_network_tweaks() {
 sub wait_for_background_process {
     my ($self, $pid, %args) = @_;
     $args{proceed_on_failure} //= 0;
-
-    my $ret = script_run("waitpid --timeout 25 $pid", %args);
-    if ($ret == 3) {    # timeout
+    # https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19853/files#r1718454521
+    my $ret = eval { script_run("wait $pid", %args); };
+    unless (defined($ret)) {
```

Comment:
This is probably nicer with 

```suggestion
    enter_cmd "wait $pid; echo wait-DONE-\$?- > /dev/$serialdev";
    unless (wait_serial "wait-DONE-0-") {
```

following the user approach as in other test code. This would use the default timeout of wait_serial which is 90s instead of the default of script_run which is 30s. If desired to use the lower timeout then add `timeout => 30` to the `wait_serial` call

Reply:
Thanks @okurz, let me try with it.

Reply:
Thx @okurz looks good.

Is there something we need to do in advance like `clear_acum` to flush the previously added output to `/dev/$serialdev` from someone else?
Or do we need to make this `wait-DONE-` more unique?

Reply:
I think it should be fine. Any previous `wait_serial` call will have flushed the previous output and it's unlikely that there are other commands in the middle that would have added the same string. If necessary one can use http://open.qa/api/testapi/#_hashed_string to make the string much more likely to be unique :)

Reply:
Hmm, seems the output was not captured by serial succeessfully.

See https://openqa.suse.de/tests/15200009#step/t11_DAD_auto4_detect/85

Reply:
`[2024-08-16T10:47:49.261580+02:00] [debug] [pid:130981] <<< testapi::wait_serial(regexp="wait-DONE-0-", no_regex=0, timeout=90, record_output=undef, expect_not_found=0, quiet=undef, buffer_size=undef)
[2024-08-16T10:47:49.262098+02:00] [debug] [pid:1004] <<< consoles::serial_screen::read_until(record_output=undef, expect_not_found=0, quiet=undef, timeout=90, no_regex=0, json_cmd_token="iXGxFVBm", pattern="wait-DONE-0-", cmd="backend_wait_serial", regexp="wait-DONE-0-", buffer_size=undef)
[2024-08-16T10:49:19.352807+02:00] [debug] [pid:130981] >>> testapi::wait_serial: wait-DONE-0-: fail`

Reply:
But it doesn't look wrong:

* https://openqa.suse.de/tests/15200009/logfile?filename=serial0.txt
  * contains `wait-DONE-0-` as the last line
* https://openqa.suse.de/tests/15200009/logfile?filename=serial_terminal.txt
  * show this command: `wait 5130; echo wait-DONE-$?- > /dev/ttyS0`

Reply:
@okurz can you plz take a look? I'm out of ideas. Do we need to switch the console before calling `wait_serial`? We are at a virtio_teriminal.

Reply:
It works as below change though:

`+    enter_cmd "wait $pid; echo wait-DONE-\$?- > /dev/$serialdev" unless is_serial_terminal();
+    enter_cmd "wait $pid; echo wait-DONE-\$?-" if is_serial_terminal();
`

http://openqa.suse.de/tests/15200246#dependencies

Reply:
ok, does that fix the issue? I would write it as

```perl
enter_cmd "wait $pid; echo wait-DONE-\$?-" . is_serial_terminal() ? '' : " > /dev/$serialdev";
```